### PR TITLE
[fix]: [PL-26379]: Fix the issue when feature flag is coming as enabled even…

### PR DESCRIPTION
**Pull request by DeepakPatankar**
… though it is disaled

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35457)

### CODE REVIEW

File: 940-feature-flag/src/main/java/io/harness/ff/FeatureFlagServiceImpl.java [hunk 1]
```diff
@@ -379,9 +379,15 @@ public void initializeFeatureFlags(DeployMode deployMode, String featureNames) {
           ? emptyList()
           : Splitter.on(',').omitEmptyStrings().trimResults().splitToList(featureNames);
```
Comment (abhinav-harness) [inline 940-feature-flag/src/main/java/io/harness/ff/FeatureFlagServiceImpl.java:380]: what happens to old record which have issues?

Comment (DeepakPatankar) [reply 940-feature-flag/src/main/java/io/harness/ff/FeatureFlagServiceImpl.java:380]: Every time the manager pods are run, the value of the feature flags is updated.

If we had an invalid feature flag (i.e. a flag which is not enabled and has accountIds), then that feature flag would also be updated and fixed at the time of startup. This feature flag would not have accountIds anymore.

**Review by Y0709 — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35457)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mitishapandey — APPROVED** [review]

**Comment by DeepakPatankar** [comment]
trigger messagecheck

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35457)* status: 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35457)* status: :shipit: complete! ✓

**Pull request merged**